### PR TITLE
Expression-based population with multiple resources

### LINKED ISSUE #644: Populate Questionnaire with Multiple Resources
**Is your feature request related to a problem? Please describe.**
This feature is the addition of populating questionnaire with not only patient resource, but also related person resource as well, as we'll have patient and related person resource available after extraction

**Describe the solution you'd like**
The approach to achieve population of Related Person(Multiple Resources) would be as follows:

1. We will pass populate function two parameters, first is questionnaire and the second parameter would be a list of Resources
2. For each Resource present in the Resource List, we will populate Initial Values for each Resource using FhirPathEngine.evaluate
3. Then We will create a Questionnaire Response, that will contain responses for all Resources, in our case it would be for Patient and Related Person

**Describe alternatives you've considered**
Currently hasn't considered any alternative

**Additional context**

**Would you like to work on the issue?**
yes I would like to work on this feature

### PR DESCRIPTION
**Pull request by ekigamba**
Fixes #644

**Description**
Implement population with multiple resources

**Alternative(s) considered**
No

**Type**
Choose one: Feature

**Screenshots (if applicable)**
N/A

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 1]
```diff
@@ -161,32 +161,40 @@ object ResourceMapper {
...
+    populateInitialValues(questionnaire.item, resources.asList())
     return QuestionnaireResponse().apply {
       item = questionnaire.item.map { it.createQuestionnaireResponseItem() }
     }
   }
 
   private suspend fun populateInitialValues(
     questionnaireItems: List<Questionnaire.QuestionnaireItemComponent>,
-    resource: Resource
+    resourceList: List<Resource>
```
[the remark this answers is not in the document]
Comment (ekigamba) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:179]: Cool

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 2]
```diff
@@ -161,32 +161,40 @@ object ResourceMapper {
...
   ) {
-    questionnaireItems.forEach { populateInitialValue(it, resource) }
+    questionnaireItems.forEach { populateInitialValue(it, resourceList) }
   }
 
   private suspend fun populateInitialValue(
     question: Questionnaire.QuestionnaireItemComponent,
-    resource: Resource
+    resourceList: List<Resource>
```
[the remark this answers is not in the document]
Comment (ekigamba) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:186]: Cool

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 3]
```diff
@@ -161,32 +161,40 @@ object ResourceMapper {
...
     if (question.type == Questionnaire.QuestionnaireItemType.GROUP) {
-      populateInitialValues(question.item, resource)
+      populateInitialValues(question.item, resourceList)
     } else {
       question.fetchExpression?.let { exp ->
-        val answerExtracted = fhirPathEngine.evaluate(resource, exp.expression)
+        val resourceType = exp.expression.substringBefore(".").removePrefix("%")
+
+        val contextResource =
+          resourceList.firstOrNull { it.resourceType.name.equals(resourceType, true) } ?: return
```
[the remark this answers is not in the document]
Comment (ekigamba) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:195]: No, I have changed this to be case-senstive

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 4]
```diff
@@ -161,32 +161,40 @@ object ResourceMapper {
...
   ) {
     if (question.type == Questionnaire.QuestionnaireItemType.GROUP) {
-      populateInitialValues(question.item, resource)
+      populateInitialValues(question.item, *resources)
     } else {
       question.fetchExpression?.let { exp ->
-        val answerExtracted = fhirPathEngine.evaluate(resource, exp.expression)
+        val resourceType = exp.expression.substringBefore(".").removePrefix("%")
+
+        val contextResource =
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:194]:
```suggestion
        // Match the first resource of the same type
        val contextResource =
```

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**